README.md
Maturaprojekt Connect-You

Gruppenmitglieder:
	Dominik Falkner
	Valentin Dorfer
	Lucas Dikany
	Martin Peneder
	Christoph Filz
	hallo foikna


ChangeLog
2013-11-26  3f69433                  [RELEASE] Release of TYPO3 6.1.6 (TYPO3 Release Team)
2013-11-26  3eda399  #53918          [BUGFIX] t3skin calls addIconSprite for each lang (Michiel Roos)
2013-11-24  93ed8d2  #51650          [BUGFIX] TS: Allow "0" as strPad.padWith (Lars Peipmann)
2013-11-24  aed6051  #15958          [BUGFIX] Reload list module on clickmenu action (Bernhard Kraft)
2013-11-21  7042298  #53802          [BUGFIX] Fix moving/copying files and folders between storages (Frans Saris)
2013-11-21  b78c694  #53844          [BUGFIX] Fix regression in ResourceCompressor (Markus Klein)
2013-11-20  3d3de05  #53243          [BUGFIX] Filemtime / Filesize trigger warning (Tomita Militaru)
2013-11-20  6c5d53d  #53458          [BUGFIX] Fluid paginate widget wrong number of links (Klaas Johan Kooistra)
2013-11-20  52b751e                  Revert "[BUGFIX] Page module: Allow to paste in empty columns" (Markus Klein)
2013-11-20  dbcaf93  #44002,#35980,  [BUGFIX] Page module: Allow to paste in empty columns (Bernhard Kraft)
2013-11-19  023014c  #38766          [BUGFIX] l10n_mode for "pages" table and group fields. (Johannes Feustel)
2013-11-19  9d97a70  #53773          [BUGFIX] Fix JS error in lang module (Markus Klein)
2013-11-19  170f084  #53750          [BUGFIX] Scheduler extension sql file is invalid (Michiel Roos)
2013-11-19  abcd5e9  #34544          [BUGFIX] fix javascript error "TBE_EDITOR not defined" in sys_action (Ralf Hettinger)
2013-11-19  ba82fac  #51998          [BUGFIX] ExtDirect StateProvider should store all settings (Johannes Feustel)
2013-11-19  571c8c9  #53746          [TASK] Optimization in AbstractViewHelper (Wouter Wolters)
2013-11-18  33b0d1b  #53707          [BUGFIX] Rename hook in VariableFrontend.php (Nicole Cordes)
2013-11-18  fbd9379  #53711          [BUGFIX] additionalAttributes for be.buttons.icon-VH misses hsc (Markus Klein)
2013-11-18  fa87ad9  #53014          [BUGFIX] Check for query failures in admin methods (Thomas Maroschik)
2013-11-15  7223b78                  Revert "[BUGFIX] EM: Fetch list as html, not as json" (Helmut Hummel)
2013-11-14  62f7e87  #45724          [BUGFIX] FILES.folder does not work (Stefan Froemken)
2013-11-14  c65640d  #51234          [BUGFIX] Move beuser property mappings to global scope (Philipp Gampe)
2013-11-14  35a95b0  #17493          [BUGFIX] Fix broken edit icons on cType HTML (Stefan Neufeind)
2013-11-13  fd66dfc  #25157,#45550   [BUGFIX] Distinguish unassigend columns and colPos 0 (Georg Ringer)
2013-11-13  0641f4f  #51918          [BUGFIX] Native date and datetime values do not consider timezone (Oliver Hader)
2013-11-12  9aa1fa2  #52926          [BUGFIX] Compressor resolves dots in filenames correctly (Christian Kuhn)
2013-11-12  fa77640  #53115          [BUGFIX] T3editor: Make errors/exceptions show correctly (Stefan Neufeind)
2013-11-12  259c64d  #22136          [BUGFIX] Fix menu popup for all IE versions (Alexander Opitz)
2013-11-12  ffd8480  #52934          [BUGFIX] dataTables: Avoid sending cookie-data too often (Stefan Neufeind)
2013-11-12  c3b0ebc  #53399          [BUGFIX] Wrong usage-text for cli_dispatch (Tomita Militaru)
2013-11-12  dcdb7bb  #52904          [BUGFIX] Evaluator in JS fails with namespaces (Stefan Aebischer)
2013-11-12  cf50919  #53538          [BUGFIX] Make be.buttons.icon-ViewHelper extensible (Stefan Neufeind)
2013-11-11  fbb19b4  #52727          [TASK] Hard-coded labels in file collections (Tomita Militaru)
2013-11-11  3dd29c3  #37948          [BUGFIX] Correctly append additionalTreelistUpdateFields (Bart Dubelaar)
2013-11-11  a3153a3  #52488          [BUGFIX] Call to FlashMessageQueue::addMessage() method in extbase (Markus Klein)
2013-11-11  b61f34f  #53423          [BUGFIX] EM: Fetch list as html, not as json (Stefan Neufeind)
2013-11-10  093d7ac  #52173          [BUGFIX] Correct storage selection (follow-up) (Ernesto Baschny)
2013-11-09  7015242  #53477          [TASK] Fix superfluous strlen() on constant strings (Steffen Ritter)
2013-11-09  827bf21  #47040          [BUGFIX] Enable treeConfig overriding by Page TSconfig (Stefan Froemken)
2013-11-09  0b03e72  #53195          [BUGFIX] T3editor: Honour fileDenyPattern on saving included TS (Stefan Neufeind)
2013-11-08  6f1625f  #29179          [BUGFIX] Escape title, extension, description of scheduler tasks (Tomita Militaru)
2013-10-23  d34bde3  #31572          [BUGFIX] Exception using cObject FORM in TypoScript (Andreas Bouche)
2013-10-18  840a3a6  #35073          [BUGFIX] Enable BE search for multiple mountpoints (Georg Ringer)
2013-10-17  775a077  #52931          [TASK] Exclude central Modernizr from concatenation (Stefan Neufeind)
2013-10-17  0382419  #52570          [TASK] Tests for Persistence\Generic\Backend::getIdentifierByObject (Stefan Neufeind)
2013-10-17  b78dc4e  #50548          [BUGFIX] Getting the identifier for a lazy object fails (Marc Bastian Heinrichs)
2013-10-16  2f1fb3f  #52529          [BUGFIX] Suppress empty tag names in output of array2xml (Markus Hoelzle)
2013-10-16  b218036  #52823          [BUGFIX] Preserve vendor name in refering request (Thomas Maroschik)
2013-10-16  88cc508                  [BUGFIX] Follow-Up: Fatal error due to missing use statement (Sascha Egerer)
2013-10-15  1761850  #52845          [BUGFIX] Moving folders fails (Oliver Hader)
2013-10-15  be9b7c7  #50802          [BUGFIX] Only load folder contents if folder is initialised (Frans Saris)
2013-10-15  ce693d8  #52824          [BUGFIX] Superfluous usage of ObjectManagerException (Oliver Hader)
2013-10-15  8be996a  #51707          [FEATURE] Add getValidators to AbstractCompositeValidator (Stefan Froemken)
2013-10-15  992e4ef  #52771          [BUGFIX] Use callback in preg_replace in RemoveXSS (Jigal van Hemert)
2013-10-14  50942c2  #52773          [BUGFIX] Detect unix-styled absolute paths on Windows systems (Nicole Cordes)

typo3/md5.js
/**
*
*  MD5 (Message-Digest Algorithm)
*  http://www.webtoolkit.info/
*
**/

function MD5(string) {

	function RotateLeft(lValue, iShiftBits) {
		return (lValue<<iShiftBits) | (lValue>>>(32-iShiftBits));
	}

	function AddUnsigned(lX,lY) {
		var lX4,lY4,lX8,lY8,lResult;
		lX8 = (lX & 0x80000000);
		lY8 = (lY & 0x80000000);
		lX4 = (lX & 0x40000000);
		lY4 = (lY & 0x40000000);
		lResult = (lX & 0x3FFFFFFF)+(lY & 0x3FFFFFFF);
		if (lX4 & lY4) {
			return (lResult ^ 0x80000000 ^ lX8 ^ lY8);
		}
		if (lX4 | lY4) {
			if (lResult & 0x40000000) {
				return (lResult ^ 0xC0000000 ^ lX8 ^ lY8);
			} else {
				return (lResult ^ 0x40000000 ^ lX8 ^ lY8);
			}
		} else {
			return (lResult ^ lX8 ^ lY8);
		}
 	}

 	function F(x,y,z) { return (x & y) | ((~x) & z); }
 	function G(x,y,z) { return (x & z) | (y & (~z)); }
 	function H(x,y,z) { return (x ^ y ^ z); }
	function I(x,y,z) { return (y ^ (x | (~z))); }

	function FF(a,b,c,d,x,s,ac) {
		a = AddUnsigned(a, AddUnsigned(AddUnsigned(F(b, c, d), x), ac));
		return AddUnsigned(RotateLeft(a, s), b);
	};

	function GG(a,b,c,d,x,s,ac) {
		a = AddUnsigned(a, AddUnsigned(AddUnsigned(G(b, c, d), x), ac));
		return AddUnsigned(RotateLeft(a, s), b);
	};

	function HH(a,b,c,d,x,s,ac) {
		a = AddUnsigned(a, AddUnsigned(AddUnsigned(H(b, c, d), x), ac));
		return AddUnsigned(RotateLeft(a, s), b);
	};

	function II(a,b,c,d,x,s,ac) {
		a = AddUnsigned(a, AddUnsigned(AddUnsigned(I(b, c, d), x), ac));
		return AddUnsigned(RotateLeft(a, s), b);
	};

	function ConvertToWordArray(string) {
		var lWordCount;
		var lMessageLength = string.length;
		var lNumberOfWords_temp1=lMessageLength + 8;
		var lNumberOfWords_temp2=(lNumberOfWords_temp1-(lNumberOfWords_temp1 % 64))/64;
		var lNumberOfWords = (lNumberOfWords_temp2+1)*16;
		var lWordArray=Array(lNumberOfWords-1);
		var lBytePosition = 0;
		var lByteCount = 0;
		while ( lByteCount < lMessageLength ) {
			lWordCount = (lByteCount-(lByteCount % 4))/4;
			lBytePosition = (lByteCount % 4)*8;
			lWordArray[lWordCount] = (lWordArray[lWordCount] | (string.charCodeAt(lByteCount)<<lBytePosition));
			lByteCount++;
		}
		lWordCount = (lByteCount-(lByteCount % 4))/4;
		lBytePosition = (lByteCount % 4)*8;
		lWordArray[lWordCount] = lWordArray[lWordCount] | (0x80<<lBytePosition);
		lWordArray[lNumberOfWords-2] = lMessageLength<<3;
		lWordArray[lNumberOfWords-1] = lMessageLength>>>29;
		return lWordArray;
	};

	function WordToHex(lValue) {
		var WordToHexValue="",WordToHexValue_temp="",lByte,lCount;
		for (lCount = 0;lCount<=3;lCount++) {
			lByte = (lValue>>>(lCount*8)) & 255;
			WordToHexValue_temp = "0" + lByte.toString(16);
			WordToHexValue = WordToHexValue + WordToHexValue_temp.substr(WordToHexValue_temp.length-2,2);
		}
		return WordToHexValue;
	};

	function Utf8Encode(string) {
		string = string.replace(/\r\n/g,"\n");
		var utftext = "";

		for (var n = 0; n < string.length; n++) {

			var c = string.charCodeAt(n);

			if (c < 128) {
				utftext += String.fromCharCode(c);
			}
			else if((c > 127) && (c < 2048)) {
				utftext += String.fromCharCode((c >> 6) | 192);
				utftext += String.fromCharCode((c & 63) | 128);
			}
			else {
				utftext += String.fromCharCode((c >> 12) | 224);
				utftext += String.fromCharCode(((c >> 6) & 63) | 128);
				utftext += String.fromCharCode((c & 63) | 128);
			}

		}

		return utftext;
	};

	var x=Array();
	var k,AA,BB,CC,DD,a,b,c,d;
	var S11=7, S12=12, S13=17, S14=22;
	var S21=5, S22=9 , S23=14, S24=20;
	var S31=4, S32=11, S33=16, S34=23;
	var S41=6, S42=10, S43=15, S44=21;

	string = Utf8Encode(string);

	x = ConvertToWordArray(string);

	a = 0x67452301; b = 0xEFCDAB89; c = 0x98BADCFE; d = 0x10325476;

	for (k=0;k<x.length;k+=16) {
		AA=a; BB=b; CC=c; DD=d;
		a=FF(a,b,c,d,x[k+0], S11,0xD76AA478);
		d=FF(d,a,b,c,x[k+1], S12,0xE8C7B756);
		c=FF(c,d,a,b,x[k+2], S13,0x242070DB);
		b=FF(b,c,d,a,x[k+3], S14,0xC1BDCEEE);
		a=FF(a,b,c,d,x[k+4], S11,0xF57C0FAF);
		d=FF(d,a,b,c,x[k+5], S12,0x4787C62A);
		c=FF(c,d,a,b,x[k+6], S13,0xA8304613);
		b=FF(b,c,d,a,x[k+7], S14,0xFD469501);
		a=FF(a,b,c,d,x[k+8], S11,0x698098D8);
		d=FF(d,a,b,c,x[k+9], S12,0x8B44F7AF);
		c=FF(c,d,a,b,x[k+10],S13,0xFFFF5BB1);
		b=FF(b,c,d,a,x[k+11],S14,0x895CD7BE);
		a=FF(a,b,c,d,x[k+12],S11,0x6B901122);
		d=FF(d,a,b,c,x[k+13],S12,0xFD987193);
		c=FF(c,d,a,b,x[k+14],S13,0xA679438E);
		b=FF(b,c,d,a,x[k+15],S14,0x49B40821);
		a=GG(a,b,c,d,x[k+1], S21,0xF61E2562);
		d=GG(d,a,b,c,x[k+6], S22,0xC040B340);
		c=GG(c,d,a,b,x[k+11],S23,0x265E5A51);
		b=GG(b,c,d,a,x[k+0], S24,0xE9B6C7AA);
		a=GG(a,b,c,d,x[k+5], S21,0xD62F105D);
		d=GG(d,a,b,c,x[k+10],S22,0x2441453);
		c=GG(c,d,a,b,x[k+15],S23,0xD8A1E681);
		b=GG(b,c,d,a,x[k+4], S24,0xE7D3FBC8);
		a=GG(a,b,c,d,x[k+9], S21,0x21E1CDE6);
		d=GG(d,a,b,c,x[k+14],S22,0xC33707D6);
		c=GG(c,d,a,b,x[k+3], S23,0xF4D50D87);
		b=GG(b,c,d,a,x[k+8], S24,0x455A14ED);
		a=GG(a,b,c,d,x[k+13],S21,0xA9E3E905);
		d=GG(d,a,b,c,x[k+2], S22,0xFCEFA3F8);
		c=GG(c,d,a,b,x[k+7], S23,0x676F02D9);
		b=GG(b,c,d,a,x[k+12],S24,0x8D2A4C8A);
		a=HH(a,b,c,d,x[k+5], S31,0xFFFA3942);
		d=HH(d,a,b,c,x[k+8], S32,0x8771F681);
		c=HH(c,d,a,b,x[k+11],S33,0x6D9D6122);
		b=HH(b,c,d,a,x[k+14],S34,0xFDE5380C);
		a=HH(a,b,c,d,x[k+1], S31,0xA4BEEA44);
		d=HH(d,a,b,c,x[k+4], S32,0x4BDECFA9);
		c=HH(c,d,a,b,x[k+7], S33,0xF6BB4B60);
		b=HH(b,c,d,a,x[k+10],S34,0xBEBFBC70);
		a=HH(a,b,c,d,x[k+13],S31,0x289B7EC6);
		d=HH(d,a,b,c,x[k+0], S32,0xEAA127FA);
		c=HH(c,d,a,b,x[k+3], S33,0xD4EF3085);
		b=HH(b,c,d,a,x[k+6], S34,0x4881D05);
		a=HH(a,b,c,d,x[k+9], S31,0xD9D4D039);
		d=HH(d,a,b,c,x[k+12],S32,0xE6DB99E5);
		c=HH(c,d,a,b,x[k+15],S33,0x1FA27CF8);
		b=HH(b,c,d,a,x[k+2], S34,0xC4AC5665);
		a=II(a,b,c,d,x[k+0], S41,0xF4292244);
		d=II(d,a,b,c,x[k+7], S42,0x432AFF97);
		c=II(c,d,a,b,x[k+14],S43,0xAB9423A7);
		b=II(b,c,d,a,x[k+5], S44,0xFC93A039);
		a=II(a,b,c,d,x[k+12],S41,0x655B59C3);
		d=II(d,a,b,c,x[k+3], S42,0x8F0CCC92);
		c=II(c,d,a,b,x[k+10],S43,0xFFEFF47D);
		b=II(b,c,d,a,x[k+1], S44,0x85845DD1);
		a=II(a,b,c,d,x[k+8], S41,0x6FA87E4F);
		d=II(d,a,b,c,x[k+15],S42,0xFE2CE6E0);
		c=II(c,d,a,b,x[k+6], S43,0xA3014314);
		b=II(b,c,d,a,x[k+13],S44,0x4E0811A1);
		a=II(a,b,c,d,x[k+4], S41,0xF7537E82);
		d=II(d,a,b,c,x[k+11],S42,0xBD3AF235);
		c=II(c,d,a,b,x[k+2], S43,0x2AD7D2BB);
		b=II(b,c,d,a,x[k+9], S44,0xEB86D391);
		a=AddUnsigned(a,AA);
		b=AddUnsigned(b,BB);
		c=AddUnsigned(c,CC);
		d=AddUnsigned(d,DD);
	}

	var temp = WordToHex(a)+WordToHex(b)+WordToHex(c)+WordToHex(d);

	return temp.toLowerCase();
}